### ISSUE TITLE
(Beta) AI Assist Feedback

### ISSUE THREAD

**Issue by MattSkewbie** [observablehq/feedback#593]
AI Assist, powered by GPT-4, for  [Pro and Enterprise](https://observablehq.com/pricing) users, can help you write SQL and generate charts quickly, accelerating you on the path to new insights. All you need to get started is an OpenAI API token.

### SQL Assistance
Writing SQL queries is an essential part of preparing data for visualization in Observable, but not everyone feels as comfortable writing SQL as they do D3 or Plot. AI Assist makes it easier than ever to write SQL from scratch, debug, and explain queries on your path to visualization. By granting access to your schema, AI Assist can craft complex, multi-table queries by simply describing the desired results. You'll spend less time digging through online documentation, and more time actually exploring your data.

### Data Analysis Pairing
Since AI Assist is built on ChatGPT, you can ask it for help during your data analysis workflow. Not sure what type of chart you should use? Send a preview of the data to AI Assist and ask it for a recommendation. It's like having a pairing partner with you at every step of the journey.

### Experimental Features
- **JavaScript Assistance**: AI Assist can help you write, debug, and explain JavaScript code. We're still training it on [Observable JavaScript](https://observablehq.com/@observablehq/observable-javascript), so pardon any incorrect results it may return.
- **Observable Plot Assistance**: AI Assist has a basic understanding of Observable Plot, and is able to generate basic charts. It also has fundamental knowledge of D3, which can help you create more complex charts.

### We're listening
If you have any feedback you would like to share about AI Assist, please let us know in the thread below.

**Comment by severo** [comment]
Are you open to adding other inference endpoints apart from OpenAI?

**Comment by MattSkewbie** [comment]
Hi @severo. Yes, we are open to adding other providers in the future. Are there any in particular that you would find useful?

**Comment by severo** [comment]
Yes, I was thinking of accessing opensource models through the [Hugging Face inference API](https://huggingface.co/docs/api-inference/quicktour#running-inference-with-api-requests)

See https://twitter.com/victormustar/status/1689224057223254016 for example

**Comment by ishaan-jaff** [comment]
Hi @severo @MattSkewbie   I’m the maintainer of LiteLLM (abstraction to call 100+ LLMs)- we allow you to create a proxy server to call 100+ LLMs, and I think it can solve your problem (I'd love your feedback if it does not)

Try it here: https://docs.litellm.ai/docs/proxy_server
https://github.com/BerriAI/litellm

## Using LiteLLM Proxy Server
```python
import openai
openai.api_base = "http://0.0.0.0:8000/" # proxy url
print(openai.ChatCompletion.create(model="test", messages=[{"role":"user", "content":"Hey!"}]))
```

## Creating a proxy server 
Ollama models
```shell
$ litellm --model ollama/llama2 --api_base http://localhost:11434
```

Hugging Face Models 
```shell
$ export HUGGINGFACE_API_KEY=my-api-key #[OPTIONAL]
$ litellm --model claude-instant-1
```


Anthropic 
```shell
$ export ANTHROPIC_API_KEY=my-api-key
$ litellm --model claude-instant-1
```
Palm

```shell
$ export PALM_API_KEY=my-palm-key
$ litellm --model palm/chat-bison
```
